snapcraft/commands/validation_sets.py

```diff
@@ -0,0 +1,166 @@
+    """Command for edit-validation-sets."""
+
+    name = "edit-validation-sets"
+    help_msg = "Edit the list of validations for <name>"
+    overview = textwrap.dedent(
+        """
+        Refer to https://snapcraft.io/docs/validation-sets for further information
+        on Validation Sets.
+        """
+    )
+
+    @overrides
+    def fill_parser(self, parser: "argparse.ArgumentParser") -> None:
+        parser.add_argument(
+            "--key-name", metavar="key-name", help="Key used to sign the assertion"
+        )
+        parser.add_argument("account_id", metavar="account-id")
+        parser.add_argument("set_name", metavar="set-name")
+        parser.add_argument("sequence", metavar="sequence")
+
+    @overrides
+    def run(self, parsed_args: "argparse.Namespace"):
+        store_client = StoreClientCLI()
+
+        asserted_validation_sets = store_client.get_validation_sets(
+            name=parsed_args.set_name, sequence=str(parsed_args.sequence)
+        )
+
+        try:
+            # assertions should only have one item since a specific
+            # sequence was requested.
+            revision = asserted_validation_sets.assertions[0].revision
+
+            snaps = yaml.dump(
+                {
+                    "snaps": [
+                        s.marshal()
+                        for s in asserted_validation_sets.assertions[0].snaps
+                    ]
+                },
+                default_flow_style=False,
+                allow_unicode=True,
+            )
+        except IndexError:
+            # If there is no assertion for a given sequence, the store API
+            # will return an empty list.
+            revision = "0"
+            snaps = _VALIDATIONS_SETS_SNAPS_TEMPLATE
+
+        unverified_validation_sets = _VALIDATION_SETS_TEMPLATE.format(
+            account_id=parsed_args.account_id,
+            set_name=parsed_args.set_name,
+            sequence=parsed_args.sequence,
+            revision=revision,
+            snaps=snaps,
+        )
+
+        with emit.pause():
+            edited_validation_sets = _edit_validation_sets(unverified_validation_sets)
+
+        if edited_validation_sets == yaml.safe_load(unverified_validation_sets):
+            emit.message("No changes made")
+        else:
+            build_assertion = store_client.post_validation_sets_build_assertion(
+                validation_sets=edited_validation_sets
+            )
+            signed_validation_sets = _sign_assertion(
+                build_assertion.marshal(), key_name=parsed_args.key_name
+            )
+            store_client.post_validation_sets(
+                signed_validation_sets=signed_validation_sets
+            )
+
+
+def _edit_validation_sets(validation_sets: str) -> Dict[str, Any]:
+    """Spawn an editor to modify the validation-sets."""
+    editor_cmd = os.getenv("EDITOR", "vi")
+
+    with tempfile.NamedTemporaryFile() as temp_file:
+        file_template_path = Path(temp_file.name)
```

Comment:
the context management here has been simplified thanks to use of Path and its write_text and read_text methods